components/src/dynamo/replay/tests/test_main.py

```
@@ -370,3 +370,24 @@ def run_synthetic(*args, **kwargs):
     assert all("nextn" not in call for call in estimator_calls)
+
+
+def test_missing_npz_planner_profile_data_is_rejected(tmp_path) -> None:
+    # The .npz short-circuit exists so replay does not import planner-only
+    # dependencies. It used to report any .npz path as already resolved,
+    # including one that is not there, and the engine answers an unreadable
+    # profile by falling back to a synthetic polynomial model rather than
+    # failing, so a typo quietly changed what was being simulated.
+    missing = tmp_path / "not-there.npz"
+
+    with pytest.raises(FileNotFoundError, match="not-there.npz"):
```

Ruff (0.16.2)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)